### PR TITLE
[niina] Narrow local variable scopes

### PR DESCRIPTION
**Pull request by peria**
BUG=#127

### CODE REVIEW

File: src/cpu/test_lockit/coma.cc [hunk 1]
```diff
@@ -339,13 +322,14 @@ int COMAI_HI::hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[], int zenkesi
...
     }
-    kesu = int(m_myf_kosuu / 55);
     if (m_aite_hakka_rensa > 3)
         m_saisoku_flag = 0;
-    saisoku = (m_saisoku_flag) && (m_myf_kosuu / 38);
 
-    for (i = 0; i < 6; i++) {
-        for (j = 0; j < 13; j++) {
+    bool shouldVanish = (m_myf_kosuu >= 55);
+    bool shouldSaisoku = (m_saisoku_flag) && (m_myf_kosuu >= 38);
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: Remove unnecessary ()

File: src/cpu/test_lockit/coma.cc [hunk 2]
```diff
@@ -300,28 +291,20 @@ int COMAI_HI::hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[], int zenkesi
...
     int yokoyose = 2;
-    int ichiren_kesi = 0;
-
-    int ee;
-    TLColor ba_ee[6][kHeight];
-    int keshiko_aa, keshiko_bb;
     int syuusoku = 0;
-
-    int ccolor, cplace, coita, cyy = 0, nidub_point_a[22] = { 0 };
+    int nidub_point_a[22] = { 0 };
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: Use {} instead of = { 0 }

File: src/cpu/test_lockit/coma.cc [hunk 3]
```diff
@@ -906,40 +882,37 @@ int COMAI_HI::hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[], int zenkesi
...
-                for (j = 0; j < 12; j++) {
-                    point[i][j] = 0;
-                }
-            }
-            num = 0;
-            for (i = 0; i < 6; i++) {
-                for (j = 0; j < 12; j++) {
+
+            int num = 0;
+            int point[6][12];
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: initialize this

File: src/cpu/test_lockit/coma.cc [hunk 4]
```diff
@@ -274,25 +274,16 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
...
 {
     TLColor ba[6][kHeight] {};
-    TLColor ba_a[6][kHeight] {};
     TLColor ba2[6][kHeight] {};
-    int point[6][12];
-    int i, j;
-    int num = 0;
-    int n;
+    TLColor ba_ee[6][kHeight];
+    TLColor ba_a[6][kHeight] {};
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: Can't we move some of these to just before calling copyField() ?
Not sure, though.

Comment (peria) [inline src/cpu/test_lockit/coma.cc]: Ah, yes.
I will split some CLs. Please wait until I'll announce again.

File: src/cpu/test_lockit/coma.cc (hunk 3 above)
Comment (peria) [inline src/cpu/test_lockit/coma.cc]: done

File: src/cpu/test_lockit/coma.cc (hunk 4 above)
Comment (peria) [inline src/cpu/test_lockit/coma.cc]: We can, but I'm not sure if it costs expensively or not.

Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: If we don't initialise it, the cost must be 0 (or almost 0. just allocating memory in stack).
But anyway, ok if it's another patch if we do so.

**Comment by mayah** [comment]
LGTM.

**Pull request merged**